crates/telio-utils/src/lru_cache.rs

```diff
@@ -157,6 +171,8 @@ impl<Key: Clone + Eq + Hash, Value> LruCache<Key, Value> {
     /// Gets the given key’s corresponding entry in the map for in-place manipulation.
     #[inline(always)]
     pub fn entry(&mut self, key: Key) -> Entry<'_, Key, Value> {
+        self.remove_expired();
```

Comment:
@Jauler pointed out this change here to me. Nice find! I've just fixed this (and updated tests to detect this) here: https://github.com/NordSecurity/libtelio/pull/222/files